### ISSUE TITLE
PLE0605 (invalid-all-format) doesn't recognize list comprehensions

### ISSUE THREAD

**Issue by eriknw** [charliermarsh/ruff#2314]
`ruff` version `v0.0.237` does not recognize list comprehension as a valid "list" type for `__all__` (rule added in https://github.com/charliermarsh/ruff/pull/2241). For example, this code:
```python
names = {"x", "y", "skip"}
__all__ = [name for name in names if name != "skip"]
```
gives
```
PLE0605 Invalid format for `__all__`, must be `tuple` or `list`
```

Note that PyLint does not recognize this as a violation.

**Comment by charliermarsh** [comment]
Added list comprehensions. I think Pylint is _even_ more permissive, but I'll adjust as issues come up.
